src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -268,16 +268,18 @@ private async Task RenegotiateAsync<TIOAdapter>(TIOAdapter adapter)
                 throw new NotSupportedException(SR.Format(SR.net_io_invalidnestedcall, nameof(WriteAsync), "write"));
             }
 
-            if (_decryptedBytesCount is not 0)
+            try
             {
-                throw new InvalidOperationException(SR.net_ssl_renegotiate_buffer);
-            }
+                if (_decryptedBytesCount is not 0)
+                {
+                    _nestedAuth = 0;
```

Comment:
I don't think we should clear it here. This was purposely left set in the finally so the RenegotiateAsync can be called at most once. 

https://github.com/dotnet/runtime/issues/49346#issuecomment-812749669

I think especially if we throw the rest would be unpredictable IMHO. 

Also if we throw for _nestedRead or _nestedWrite this would still be set. I think that is ok as the intention is not to allow multiple calls but it would be inconsistent.

Reply:
You are right, I was tempted by wrong change in test.